PROVESFlightControllerReference/Components/TcSecurityDeframer/TcSecurityDeframer.cpp

```
@@ -186,28 +209,78 @@ void TcSecurityDeframer ::configure() {
+    value = storedValue;
+    return Os::File::Status::OP_OK;
+}
+
+void TcSecurityDeframer ::prepareForReboot_handler(FwIndexType portNum) {
+    // Planned reboot: persist the EXACT current sequence number, not the write-ahead
+    // high-water mark. On the next boot the counter resumes at precisely the last
+    // accepted value, so ground (at lastAccepted + 1) stays inside the acceptance
+    // window with no resync needed. Unplanned reboots (crash/power loss) still resume
+    // from the write-ahead mark -- that direction is the security-conservative one.
+    Os::ScopeLock lock(this->m_sequenceNumberLock);
+    const Os::File::Status status = this->writeSequenceNumber(this->m_sequenceNumber);
```

Clang (14.0.6)

[warning] variable 'status' is not initialized

(cppcoreguidelines-init-variables)